File: FlinkJavaCode/src/main/java/com/wubaibao/flinkjava/code/chapter3/FlinkLocalWebUI.java
package com.wubaibao.flinkjava.code.chapter3;

import org.apache.flink.api.common.typeinfo.Types;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.configuration.Configuration;
import org.apache.flink.configuration.RestOptions;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.util.Collector;

/**
 * 测试 Flink 本地WebUI
 */
public class FlinkLocalWebUI {
    public static void main(String[] args) throws Exception {
        //1.使用本地模式
        Configuration conf = new Configuration();
        //设置WebUI绑定的本地端口
        conf.setString(RestOptions.BIND_PORT,"8081");
        //使用配置
        StreamExecutionEnvironment env = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(conf);

        //2.读取Socket数据
        DataStreamSource<String> ds = env.socketTextStream("node5", 9999);

        //3.准备K,V格式数据
        SingleOutputStreamOperator<Tuple2<String, Integer>> tupleDS = ds.flatMap((String line, Collector<Tuple2<String, Integer>> out) -> {
            String[] words = line.split(",");
            for (String word : words) {
                out.collect(Tuple2.of(word, 1));
            }
        }).returns(Types.TUPLE(Types.STRING, Types.INT));

        //4.聚合打印结果
        tupleDS.keyBy(tp -> tp.f0).sum(1).print();

        //5.execute触发执行
        env.execute();
    }
}

File: FlinkScalaCode/src/main/scala/com/wubaibao/flinkscala/code/chapter3/FlinkLocalWebUI.scala
package com.wubaibao.flinkscala.code.chapter3

import org.apache.flink.configuration.{Configuration, RestOptions}
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment

/**
 * Scala 使用Flink本地WebUI
 */
object FlinkLocalWebUI {
  def main(args: Array[String]): Unit = {

    //1.创建本地WebUI环境
    val configuration = new Configuration()
    //设置绑定的本地端口
    configuration.set(RestOptions.BIND_PORT,"8081")
    //第一种设置方式
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(configuration)

    //2.Scala 流处理导入隐式转换，使用Scala API 时需要隐式转换来推断函数操作后的类型
    import org.apache.flink.streaming.api.scala._

    //3.读取Socket数据
    val linesDS: DataStream[String] = env.socketTextStream("node5", 9999)

    //4.进行WordCount统计
    linesDS.flatMap(line=>{line.split(",")})
      .map((_,1))
      .keyBy(_._1)
      .sum(1)
      .print()

    //5.最后使用execute 方法触发执行
    env.execute()

  }

}
